### PR TITLE
Add builder dev doc

### PR DESCRIPTION
**Pull request by pozsgaic**
Signed-off-by: pozsgaic <chris.pozsgai@progress.com>

### CODE REVIEW

File: DEBUGGING.md [hunk 1]
```diff
@@ -0,0 +1,625 @@
...
+   
+This section discusses mainly how to build the Rust based services, but does minimally discuss Builder UI building.  
+
+### Linux Environment
+When building builder, each builder service is built separately.
+Typically, you'll want to clone the repo https://github.com/habitat-sh/builder.git on both the Mac host and on the VM you plan to use for the backend.  This will enable you to run
+Linux builds inside the VM and run the Builder UI from the front-end 
+
+### MacOS Environment
+For the front-ends, kicking off a build is as easy as typing `npm start` from the components/builder-web directory.  Implicit in the
```
Comment (rahulgoel1) [inline DEBUGGING.md:38]: It seems like you wanted to add more details but missed it with the statement "Implicit in the"

**Review by rahulgoel1 — COMMENTED** [review]
Looks good , I think there is one comment which I had added.

File: DEBUGGING.md [hunk 2]
```diff
@@ -0,0 +1,625 @@
...
+
+From outside of the studio on VM host Linux terminal:
+
+`$` `curl -s http://localhost:9631/services/builder-api/default | jq -r .cfg`
+
+You will see a section [datastore], and the metadata for the database are defined in that section.  Of most importance is the password.  Copy into the clipboard or set as environment variable $PGPASSWORD
+
+Log into the database:
+
+`$` `psql -d builder -U hab -W`
```
Comment (dikshagupta1) [inline DEBUGGING.md:276]: Please change the "$" symbol to "#"

**Comment by pozsgaic** [comment]
We will move this doc to an internal repository once we establish the location.

**Comment by pozsgaic** [comment]
Content has been moved to https://github.com/habitat-sh/internal_dev_docs

**Pull request closed without merging**